### PR TITLE
feat(#8644): Support concurrency for haproxy-healthcheck

### PR DESCRIPTION
**Pull request by nydr**
# Description
PR moved from https://github.com/medic/cht-core/pull/8733 to update branch name

Sticking with python to reuse existing code as much as possible

Addressing my main concerns for this app

Supports more than one concurrent connection
originally this app could hang if a connection wasn't closed properly
Package dependencies in container and freeze versions
originally dependencies would be installed on startup and install the latest version, this could cause issues as a pypi outage or an incompatible release of dependent package would prevent an already deployed app to start
I added some basic tests but the coverage has room for improvement, since this service has limited lifetime I wanted to check for second opinions if more tests are needed before I spend more time on it

Related
- https://github.com/medic/cht-core/issues/8644

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8644-concurrent-haproxy-healthcheck/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8644-concurrent-haproxy-healthcheck/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8644-concurrent-haproxy-healthcheck/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Review by dianabarsan — APPROVED** [review]
I think this is great! My only residual comment is about log levels, because I believe setting the default log level to info produces a lot of noise in logs that nobody will be interested in seeing (or storing).

**Comment by garethbowen** [comment]
@nydr I'd love to get this merged and released in 4.6.0 and I see it's approved. Is there anything else you're waiting on before merging?

**Comment by nydr** [comment]
Updated default log level, I also set the initialization log entry to warning as I think it's important to see that by default even if it technically shouldn't be a warning 😄

File: haproxy-healthcheck/check.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ async def main() -> None:
     )
 
     addrs: str = ", ".join(str(sock.getsockname()) for sock in server.sockets)
-    log.info(f"Serving on {addrs}")
+    log.warning(f"Serving on {addrs}")
```
Comment (dianabarsan) [inline haproxy-healthcheck/check.py:87]: Devil in the details 😆

**Review by dianabarsan — APPROVED** [review]
Let's ship it!

**Pull request merged**